content/knowledge/scalinghistory.md

```diff
@@ -0,0 +1,135 @@
+   This is the problem when there are various nodes communicating with each other and a few of them are “Byzantine” which means either faulty or malicious.
+   A good consensus protocol should be able to tackle this issue in a decentralised manner for better security of the network.
+   One of the consensus algorithms that manage to do it is called “Federated Byzantine Agreement”.
+   This was introduced by Stellar.
+   In FBA, every node has its “quorum set” which is a list of other nodes which this particular node trusts.
+   Therefore to achieve consensus for a transaction, the node relies on the members in its quorum set.
+   As long as the majority of nodes are not malicious, this will ensure security of the system.
+   Complex web of overlapping quorum slices makes it almost impossible for the majority of nodes to collude to control consensus.
+   FBA can be used on top of PoW or PoS (to be discussed later).
+   Safety and fault tolerance are [preferred over liveness](https://www.youtube.com/watch?v=aU08km2xrz0&ab_channel=Lumenauts).
+   In case of an accidental fork, the system is halted until consensus is reached.
+   This is important in case of banking applications.
+   Consensus that requires only message passing followed by voting process leads to high transaction volume per second and less expenditure of electricity than PoW.
+   However, a criticism of pure FBA is that it leads to fragile structures of constituent nodes, permitting topology scenarios where a single node failure can cause a network-wide failure.
+
+   TODO: Drawback: Single point failure if quorum sets share a single node?
+   (Not mentioning about BFT which require a pre recommended validator list as it is against decentralisation)
+
+## 3. Proof of Stake (PoS)
+
+   Then came the Proof of Stake(PoS) consensus algorithms (e.g. Casper).
+   This no longer needs the computational complexity for validation of blocks but rather how much a node or node pool has staked and how long it has been active.
+   Staking is like putting some native cryptocurrency as a collateral.
+   The higher the stake and time in the game, the higher the chances of getting to write a block on behalf of the whole ecosystem and in return receive some cryptocurrency.
+   This can be distributed among the participating nodes in case of a node pool where there are multiple parties participating by delegating their tokens.
+   Two keywords are different here compared to PoW: (1) Computers that participate in PoS consensus are called Validators, (2) Once a block has been created and accepted by the network, it is said to be forged (not mined).
+   Then the validator gets some reward.
+   If a validator tries to be malicious then it gets penalised losing some of its staked cryptocurrency.
+   PoS is much faster than PoW as there is no need to solve complex cryptographic puzzles to get to be the block writer and the process is much more streamlined.
```

Comment:
It's not much faster. First of all, PoW miners generally use specialized hardware that does only hashing, which is completely useless for any other kind of work, so it wouldn't help with executing transactions or verifying signatures. Second, PoS has a higher communication complexity, so the bottleneck is the network's ability to communicate about the consensus. In PoW, the network just needs to receive a block and can verify it without needing to request anyone's opinion on it, as its implicitly valid if it follows the consensus rules.

Reply:
Got it. Thanks. So I shouldn't make any statement regarding which one is faster as it will depend on specific cases? Or is it still fair to say `PoS is relatively faster`?